checker/tests/optional/ShadowingMethodCalls.java

```diff
@@ -43,4 +44,29 @@ void callThisAndCheckThis() {
       super.f.get();
     }
   }
+
+  static class ClassA {
+
+    @Pure
+    Optional<String> getOpt() {
+      throw new RuntimeException();
+    }
+  }
+
+  static class ClassB extends ClassA {
+    void use() {
+      if (super.getOpt().isPresent()) {
+        String s = super.getOpt().get();
+      }
+      if (super.getOpt().isPresent()) {
+        String s = this.getOpt().get();
+      }
+      if (this.getOpt().isPresent()) {
+        String s = super.getOpt().get();
+      }
+      if (this.getOpt().isPresent()) {
+        String s = this.getOpt().get();
+      }
+    }
+  }
```

Comment:
Thanks @smillst! I believe I fixed this failing test case via [this commit](https://github.com/typetools/checker-framework/pull/6611/commits/27a9a17e4b9bd84ba1954b4c4a3467d7289fb1d9)